src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectFactory.Windows.cs

```diff
@@ -8,29 +8,43 @@ namespace System.Data.SqlClient
 {
     internal sealed class TdsParserStateObjectFactory
     {
-        public static bool useManagedSni = false;
+
+        private const string UseLegacyNetworkingOnWindows = "System.Data.SqlClient.UseLegacyNetworkingOnWindows";
 
         public static readonly TdsParserStateObjectFactory Singleton = new TdsParserStateObjectFactory();
 
+        static TdsParserStateObjectFactory()
+        {
+            bool shouldUseLegacyNetorking;
+            // If The appcontext switch is set then Use Managed SNI based on the value. Otherwise Managed SNI should always be used.
+            UseManagedSNI = AppContext.TryGetSwitch(UseLegacyNetworkingOnWindows, out shouldUseLegacyNetorking) ? !shouldUseLegacyNetorking : true;
+        }
```

Comment:
(Plus having the result stored in a readonly static bool will potentially allow the JIT to do further optimizations, essentially treating the value as a const at most usage sites and optimizing the compilation of those locations as if a const were used, e.g. dead branch elimination.)